src/photon/photon_scheduler.h

```diff
@@ -4,6 +4,9 @@
 #include "task.h"
 #include "event_loop.h"
 
+/* The duration between local scheduler heartbeats. */
+#define LOCAL_SCHEDULER_HEARTBEAT_TIMEOUT_MILLISECONDS 1000
```

Comment:
one heartbeat per second is going to be ok for checking liveness, but too infrequent for available_workers information to be actionable. Part of the issue here is that we should really be running these updates in a separate thread with read-only access to the queue information.